addons/binding/org.openhab.binding.amazonechocontrol/src/main/java/org/openhab/binding/amazonechocontrol/internal/Connection.java

```diff
@@ -0,0 +1,925 @@
+        Gson gson = new Gson();
+
+        JsonObject operationPayload = new JsonObject();
+        operationPayload.addProperty("deviceType", device.deviceType);
+        operationPayload.addProperty("deviceSerialNumber", device.serialNumber);
+        operationPayload.addProperty("locale", "");
+        operationPayload.addProperty("customerId", device.deviceOwnerCustomerId);
+        if (parameters != null) {
+            for (String key : parameters.keySet()) {
+                Object value = parameters.get(key);
+                if (value instanceof String) {
+                    operationPayload.addProperty(key, (String) value);
+                } else if (value instanceof Number) {
+                    operationPayload.addProperty(key, (Number) value);
+                } else if (value instanceof Boolean) {
+                    operationPayload.addProperty(key, (Boolean) value);
+                } else if (value instanceof Character) {
+                    operationPayload.addProperty(key, (Character) value);
+                } else {
+                    operationPayload.add(key, gson.toJsonTree(value));
+                }
+            }
+        }
+
+        JsonObject startNode = new JsonObject();
+        startNode.addProperty("@type", "com.amazon.alexa.behaviors.model.OpaquePayloadOperationNode");
+        startNode.addProperty("type", command);
+        startNode.add("operationPayload", operationPayload);
+
+        JsonObject sequenceJson = new JsonObject();
+        sequenceJson.addProperty("@type", "com.amazon.alexa.behaviors.model.Sequence");
+        sequenceJson.add("startNode", startNode);
+
+        JsonStartRoutineRequest request = new JsonStartRoutineRequest();
+        request.sequenceJson = gson.toJson(sequenceJson);
+        String json = gson.toJson(request);
+
+        makeRequest("POST", alexaServer + "/api/behaviors/preview", json, true, true, null);
+    }
+
+    public void startRoutine(Device device, String utterance) throws IOException, URISyntaxException {
+        JsonAutomation found = null;
+        String deviceLocale = "";
+        for (JsonAutomation routine : getRoutines()) {
+            Trigger[] triggers = routine.triggers;
+            if (triggers != null && routine.sequence != null) {
+                for (JsonAutomation.Trigger trigger : triggers) {
+                    if (trigger == null) {
+                        continue;
+                    }
+                    Payload payload = trigger.payload;
+                    if (payload == null) {
+                        continue;
+                    }
+                    if (StringUtils.equalsIgnoreCase(payload.utterance, utterance)) {
+                        found = routine;
+                        deviceLocale = payload.locale;
+                        break;
+                    }
+                }
+            }
+        }
+        if (found != null) {
+            Gson gson = new Gson();
```

Comment:
This is potentially expensive - you should instantiate it only once as a field in that class.